examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,397 @@
+# completely at random (MCAR), i.e. the missingness does not depend on the
+# observed data or the unobserved data. We can simulate such scenario by
+# randomly replacing values from randomly selected features with `Nan` values.
+
+import numpy as np
+
+from sklearn.model_selection import TimeSeriesSplit, cross_validate
+
+np.random.seed(42)
+
+ts_cv = TimeSeriesSplit(n_splits=5, gap=48, max_train_size=10000, test_size=1000)
+train_0, test_0 = next(ts_cv.split(df))
+last_days = slice(-192, None)
+total_cells = X.shape[0] * X.shape[1]
+missing_fraction_list = [0, 0.01, 0.03]
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y.iloc[test_0].values[last_days], label="Actual transfer")
+
+for missing_fraction in missing_fraction_list:
+    num_missing_cells = int(total_cells * missing_fraction)
+    row_indices = np.random.choice(X.shape[0], num_missing_cells, replace=True)
+    col_indices = np.random.choice(X.shape[1], num_missing_cells, replace=True)
+    X = df.drop(columns=["transfer", "class"])
+    X.iloc[row_indices, col_indices] = np.nan
+
+    hgbt.fit(X.iloc[train_0], y.iloc[train_0])
+    hgbt_predictions = hgbt.predict(X.iloc[test_0])
+    cv_results = cross_validate(
+        hgbt,
+        X,
+        y,
+        cv=ts_cv,
+        scoring="neg_root_mean_squared_error",
+    )
+    rmse = -cv_results["test_score"]
+    ax.plot(
+        hgbt_predictions[last_days],
+        label=(
+            f"missing_fraction={missing_fraction}, RMSE={rmse.mean():.2f} +/-"
+            f" {rmse.std():.2f}"
+        ),
+        alpha=0.5,
+    )
+ax.set(
+    title="Daily energy transfer predictions on data with MCAR values",
+    xticks=[(i + 0.25) * 48 for i in range(4)],
+    xticklabels=["Tue", "Wed", "Thu", "Fri"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend()
+
+# %%
+# Support for quantile loss
+# =========================
+#
+# The quantile loss in regression enables a view of the variability or
+# uncertainty of the target variable. For instance, predicting the 5th and 95th
+# percentiles can provide a 90% prediction interval, i.e. the range within which
+# we expect the true value to fall with 90% probability.
+
+from sklearn.metrics import make_scorer, mean_pinball_loss
+
+quantiles = [0.95, 0.05]
+predictions = []
+X = df.drop(columns=["transfer", "class"])  # reset X
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y.iloc[test_0].values[last_days], label="Actual transfer")
+
+for quantile in quantiles:
+    hgbt_quantile = HistGradientBoostingRegressor(
+        loss="quantile", quantile=quantile, **common_params
+    )
+    hgbt_quantile.fit(X.iloc[train_0], y.iloc[train_0])
+    hgbt_predictions = hgbt_quantile.predict(X.iloc[test_0])
+
+    predictions.append(hgbt_predictions)
+    cv_results = cross_validate(
+        hgbt_quantile,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=make_scorer(mean_pinball_loss, alpha=quantile),
```

Comment:
If I am not wrong, this is the same here. We indicate a score that does not corresponds to the values seen in the plot.